concepts/traits/about.md

```diff
@@ -1 +1,315 @@
+    use super::ShapeGeometry;
+
+    #[derive(Copy, Drop)]
+    pub struct Rectangle {
+        pub height: u64,
+        pub width: u64,
+    }
+
+    // Implementation RectangleGeometry passes in <Rectangle>
+    // to implement the trait for that type
+    impl RectangleGeometry of ShapeGeometry<Rectangle> {
+        fn boundary(self: Rectangle) -> u64 {
+            2 * (self.height + self.width)
+        }
+        fn area(self: Rectangle) -> u64 {
+            self.height * self.width
+        }
+    }
+}
+
+mod circle {
+    // Importing ShapeGeometry is required to implement this trait for Circle
+    use super::ShapeGeometry;
+
+    #[derive(Copy, Drop)]
+    pub struct Circle {
+        pub radius: u64
+    }
+
+    // Implementation CircleGeometry passes in <Circle>
+    // to implement the imported trait for that type
+    impl CircleGeometry of ShapeGeometry<Circle> {
+        fn boundary(self: Circle) -> u64 {
+            (2 * 314 * self.radius) / 100
+        }
+        fn area(self: Circle) -> u64 {
+            (314 * self.radius * self.radius) / 100
+        }
+    }
+}
+
+use rectangle::Rectangle;
+use circle::Circle;
+
+fn main() {
+    let rect = Rectangle { height: 5, width: 7 };
+    println!("Rectangle area: {}", ShapeGeometry::area(rect)); //35
+    println!("Rectangle boundary: {}", ShapeGeometry::boundary(rect)); //24
+
+    let circ = Circle { radius: 5 };
+    println!("Circle area: {}", ShapeGeometry::area(circ)); //78
+    println!("Circle boundary: {}", ShapeGeometry::boundary(circ)); //31
+}
+```
+
+The implementation visibility for both the `CircleGeometry` and `RectangleGeometry` does not need to be public since the `ShapeGeometry` trait is made public and should be easily found by the compiler.
+
+## Impl Aliases
+
+Implementation, when imported, can be labeled as `impl` as an alternative name. This helps when instantiating a generic implementation for concrete types. Let's define a trivial `Two` implementation for all types that implement the `One` trait by simply adding twice the value of one and returning the summation. We can afterwards make available for use an implementation that works for u8 and u128 types.
+
+```rust
+trait Two<T> {
+    fn two() -> T;
+}
+
+mod one_based {
+    pub impl TwoImpl<
+        T, +Copy<T>, +Drop<T>, +Add<T>, impl One: core::num::traits::One<T>
+    > of super::Two<T> {
+        fn two() -> T {
+            One::one() + One::one()
+        }
+    }
+}
+
+pub impl U8Two = one_based::TwoImpl<u8>;
+pub impl U128Two = one_based::TwoImpl<u128>;
+```
+
+The generic implementation can be written in a private module and with the use of impl alias, we can instantiate the generic implementation of these two concrete types. The benefit of using a private and unexposed generic implementation is that it prevents code duplication across files. This as well keeps the public API simple and clean.
+
+## Negative Implementation
```

Comment:
Let's describe only the production-ready features of Cairo, not experimental ones. 
**Remove this section for now**, and we can add it back once Negative Impls are part of the core language. 

Created an issue to track this https://github.com/exercism/cairo/issues/143